app/assets/javascripts/discourse/app/components/modal/discard-draft.hbs

```diff
@@ -0,0 +1,28 @@
+<DModal @closeModal={{@dismissModal}} class="discard-draft-modal">
```

Comment:
There is no `dismissModal` argument, so I think you meant either `this.dismissModal` or `@closeModal`?